src/Functions/Public/iaas/Get-vRAvSphereFabricNetwork.ps1

```diff
@@ -1,121 +1,166 @@
+        [ValidateNotNullOrEmpty()]
+        [String[]]$Id,
+
+        [Parameter(Mandatory=$true,ParameterSetName="ByName")]
+        [ValidateNotNullOrEmpty()]
+        [String[]]$Name,
+
+        [Parameter(Mandatory=$false)]
+        [ValidateNotNullOrEmpty()]
+        [String[]]$CloudAccountName,
+
+        [Parameter(Mandatory=$false)]
+        [ValidateNotNullOrEmpty()]
+        [String[]]$CloudAccountId
+
+    )
+    Begin {
+
+        $APIUrl = "/iaas/api/fabric-networks-vsphere"
+
+        function CalculateOutput([PSCustomObject]$FabricNetworks) {
+
+            foreach ($FabricNetwork in $FabricNetworks.content) {
+                [PSCustomObject]@{
                     ExternalRegionId = $FabricNetwork.externalRegionId
-                    CloudAccountIds = $FabricNetwork.cloudAccountIds
+                    DefaultIpv6Gateway = $FabricNetwork.defaultIpv6Gateway
+                    Description = $FabricNetwork.description
                     ExternalId = $FabricNetwork.externalId
+                    DefaultGateway = $FabricNetwork.defaultGateway
+                    OrgId = $FabricNetwork.orgId
+                    Tags = $FabricNetwork.tags
+                    OrganizationId = $FabricNetwork.organizationId
+                    CreatedAt = $FabricNetwork.createdAt
+                    Ipv6Cidr = $FabricNetwork.ipv6Cidr
+                    CloudAccountIds = $FabricNetwork.cloudAccountIds
+                    IsDefault = $FabricNetwork.isDefault
+                    Domain = $FabricNetwork.domain
+                    DnsServerAddresses = $FabricNetwork.dnsServerAddresses
                     Name = $FabricNetwork.name
+                    IsPublic = $FabricNetwork.isPublic
+                    Cidr = $FabricNetwork.cidr
                     Id = $FabricNetwork.id
-                    CreatedAt = $FabricNetwork.createdAt
                     UpdatedAt = $FabricNetwork.updatedAt
-                    OrganizationId = $FabricNetwork.organizationId
-                    Links = $FabricNetwork._links
+                    DnsSearchDomains = $FabricNetwork.dnsSearchDomains
                 }
             }
         }
-    
-        process {
-    
-            try {
-    
-                switch ($PsCmdlet.ParameterSetName) {
-    
-                    # --- Get Fabric Network by Id
-                    'ById' {
-    
-                        foreach ($FabricNetworkId in $Id){
-    
-                            $URI = "$($APIUrl)?`$filter=id eq '$($FabricNetworkId)'"
-                            $Response = Invoke-vRARestMethod -Method GET -URI $URI -Verbose:$VerbosePreference
-    
-                            foreach ($FabricNetwork in $Response.content) {
-                                CalculateOutput $FabricNetwork
-                            }
-                        }
-    
-                        break
+
+        function buildAccountQuery() {
```

Comment:
I think this function should have parameters for ```$CloudAccountName``` and ```$CloudAccountId``` rather than rely on them being available from a higher scope with those names.

Reply:
This is changed in latest and tested.